main/zoe/api/zoe-contract-facet.md

```diff
@@ -166,23 +164,36 @@ Zoe.
 const { issuerKeywordRecord, keywords, terms } = zoe.getInstanceRecord()
 ```
 
-## zcf.getCurrentAllocation(offerHandle, sparseKeywords)
+## zcf.getCurrentAllocation(offerHandle, brandKeywordRecord)
 - `offerHandle` <router-link to="/glossary/#handle">`{Array <Handle>}`</router-link>
-- `sparseKeywords` sparseKeywords is an array of string keywords, which may be a subset of allKeywords.
-- Returns: <router-link to="/zoe/api/records.html#amount-keyword-record">`{<AmountKeywordRecord>}`</router-link>
+- `brandKeywordRecord` <router-link to="/zoe/api/records.html#brand-keyword-record">
+`{[<BrandKeywordRecord>]}`</router-link> is an optional record mapping keywords to brands for each of the offers.
+- Returns: <router-link to="/zoe/api/records.html#amount-keyword-record">
+`{<AmountKeywordRecord>}`</router-link>
+
+For the keywords specified by `brandKeywordRecord, get the amounts currently allocated to the offer. If
+`brandKeywordRecord` is not specified, amounts are returned for only the Keywords that currently have
+allocated amounts (including empty amounts.) This might mean amounts that will be included in the
+payout could be omitted if not specified.
 
-Get the amounts associated with the sparseKeywords for the offer.
+If `brandKeywordRecord` is specified, for keywords without amounts assigned to the offer, empty amounts are filled in 
 
 ```js
 const { foo, bar } = zoe.getCurrentAllocation(offerHandle, ['foo', 'bar']);
 ```
 
-## zcf.getCurrentAllocations(offerHandles, sparseKeywords)
+## zcf.getCurrentAllocations(offerHandles, brandKeywordRecords)
 - `offerHandles` <router-link to="/glossary/#handle">`{Array <Handle>}`</router-link>
-- `sparseKeywords` sparseKeywords is an array of string keywords, which may be a subset of allKeywords.
+- `brandKeywordRecords` <router-link to="/zoe/api/records.html#brand-keyword-record">
+`{[ Array <BrandKeywordRecords>]}`</router-link> `brandKeywordRecords` is an optional array of records
+mapping keywords to brands for each offer.
 - Returns: <router-link to="/zoe/api/records.html#amount-keyword-record">`{[<AmountKeywordRecord>]}`</router-link>
 
-Get a list of the amounts associated with the sparseKeywords for the offers.
+For each `brandKeywordRecord specified keyword, from each offer get its currently allocated amount. If
+`brandKeywordRecord``` is not specified, amounts are returned only for the Keywords that currently have
+allocated amounts (including empty amounts.) Amounts included in the payout might be omitted if not specified.
+
+If `brandKeywordRecord` is specified, and some keywords' amounts haven't been assigned to the offer, empty amounts are filled in.
```

Comment:
The backticks here aren't right. Please take another look.

I like the phrasing on line 174 better.

This one says " for each X, from each Y, get Z."
The earlier one says "for each X, get the Z from the Y."

When we get the wording right, the variants in zcf should match those in zoe. Both files should have a `getCurrentAllocation()` and `getCurrentAllocations()` that differ only in the pluralization of both parameters in three places: 
 * index line
 * parameter break-out list
 * prose description.